### ISSUE TITLE
Sleep timer

### ISSUE THREAD

**Issue by ToNIX-** [vanilla-music/vanilla#157]
Any plans on adding a simple sleep timer in Vanilla Music (Menu > Sleep timer)?

I know it can be achieved with the "Idle Timeout" option in Settings > Miscellaneous.

However, once activated, the "Idle Timeout" stays active forever, unless unchecked. I think that adding an option that would deactivate the "Idle Timeout" once done would make sense. The checkbox could be name something like "Disable once done" and grayed out unless "Idle Timeout" is checked.

Thanks :)

**Comment by adrian-bl** [comment]
Thinking about it, i feel like this should be the default anyway: I often find myself having the idle-timeout active by mistake.

So i'll have the idle-timeout automatically disable itself if there are no objections

**Comment by Allanitomwesh** [comment]
It could just disable itself after it reaches the timeout. That would be a sleep timer already. Then the checkbox would be "Always Timeout" and greyed out,that would maintain current behaviour

**Comment by ToNIX-** [comment]
Looks good to me, it should at least disable itself automatically once it reaches the timeout.

**Comment by ToNIX-** [comment]
Any news on this one? I tried to play in the code but hmmm. Haven't touched programming since almost 10 years, let alone Android programming. I do somewhat understand how it works, but can't reset the use_idle_timeout preference.

**Comment by adrian-bl** [comment]
No, sorry: i did not have time to work on this.

> but can't reset the idle_timeout preference.

The code should get an instance of the SharedPreferencesEditor after the idle timeout was hit, eg:
`
SharedPreferences.Editor editor = PlaybackService.getSettings(this).edit();
editor.putBoolean(PrefKeys.USE_IDLE_TIMEOUT, false);
editor.commit();
`

**Comment by M1Aston** [comment]
Yes, please add a real sleep timer. The timeout option is too limited.  Other players have option like
- stop playing after x number of files.
- stop after x hours\minutes.
- stop after x hours\minutes but always finish the playing file.
- close the program instead after stopping playback.

Please improve the sleep timer.

**Comment by M1Aston** [comment]
Yes, please add a a real sleep timer, with (some of) the options mentioned above. :-)

**Comment by gappleto97** [comment]
Is this a planned feature still?
